lib/rules/string.js

```diff
@@ -0,0 +1,407 @@
+const hostnameRegEx = /^(([a-z\d]|[a-z\d][a-z\d\-]*[a-z\d])\.)*([a-z\d]|[a-z\d][a-z\d\-]*[a-z\d])$/i;
+/* eslint-enable max-len */
+
+class StringValidator extends SyncRule {
+    validateSync(params) {
+        return typeof params.value === 'string' &&
+            params.value.length > 0;
+    }
+
+    static ruleName() {
+        return 'string';
+    }
+}
+
+/* eslint-disable no-unreachable */
+class Insensitive extends Rule {
+    compile(params) {
+        function cmp(a, b) {
+            return typeof a === 'string' && typeof b === 'string' &&
+                a.toLowerCase() === b.toLowerCase();
+        }
+
+        params.invokeAll(ComparatorRule, rule => rule.addComparator(cmp));
+    }
+
+    static ruleName() {
+        return 'string.insensitive';
+    }
+}
+/* eslint-enable ignore-unreachable */
+
+
+class Min extends SyncRule {
+    compile(params) {
+        this._val = params.args[0];
+    }
+
+    validateSync(params) {
+        return params.value.length >= this._val || {
+            length: params.value.length,
+            min: this._val,
+        };
+    }
+
+    static ruleName() {
+        return 'string.min';
+    }
+}
+
+class Max extends SyncRule {
+    compile(params) {
+        this._val = params.args[0];
+    }
+
+    validateSync(params) {
+        return params.value.length <= this._val || {
+            length: params.value.length,
+            max: this._val,
+        };
+    }
+
+    static ruleName() {
+        return 'string.max';
+    }
+}
+
+class Length extends SyncRule {
+    compile(params) {
+        this._val = params.args[0];
+    }
+
+    validateSync(params) {
+        return params.value.length === this._val || {
+            length: params.value.length,
+            expected: this._val,
+        };
+    }
+
+    static ruleName() {
+        return 'string.length';
+    }
+}
+
+class CreditCard extends SyncRule {
+    validateSync(params) {
+        const str = params.value;
+        if (!ccRegex.test(str)) {
+            return false;
+        }
+        let sum = 0;
+        for (let i = 0; i < str.length; i++) {
+            let intVal = parseInt(str.substr(i, 1), 10);
+            if (i % 2 === 0) {
+                intVal *= 2;
+                if (intVal > 9) {
+                    intVal = 1 + (intVal % 10);
+                }
+            }
+            sum += intVal;
+        }
+        return (sum % 10) === 0;
```

Comment:
It'd be preferable to just import it from [isCreditCard](https://github.com/chriso/validator.js/blob/master/src/lib/isCreditCard.js), I don't see a reason to duplicate the code here. If you duplicate it you also have to include the license from Validator.js.